src/util.cpp

```diff
@@ -840,6 +842,10 @@ static bool GetConfigOptions(std::istream& stream, std::string& error, std::vect
             } else if ((pos = str.find('=')) != std::string::npos) {
                 std::string name = prefix + TrimString(str.substr(0, pos), pattern);
                 std::string value = TrimString(str.substr(pos + 1), pattern);
+                if (used_hash && name == "rpcpassword") {
+                    error = strprintf("parse error on line %i, illegal hash character used in rpcpassword", linenr);
```

Comment:
concept ACK—
though please reword this message, the `#` is not so much *illegal*, it starts a legal comment; it's just that *probably* the user made a mistake, assuming the `#`  will actually be part of the password

Reply:
Fixed :)